### ISSUE TITLE
heading ids may produce invalid html

### ISSUE THREAD

**Issue by snscltt** [Ypnose/ahrf#3]
In the case of naming two or more headings the same, this will produce invalid html.
Maybe the most simple way to avoid this would be to append an incrementing number?

This might be a very simple fix:
https://github.com/savoir-vivre/ahrf/commit/990414e41477cdfb9cdf0042810d4527e6bfc29b

**Comment by Ypnose** [comment]
Do you have an example where two or more headings have the same title? Appending a "counter" after every titles is kinda broken, IMHO.

**Comment by alexh-name** [comment]
I had it somewhere in the wild. I don't remember where.
I think in most cases good choices in headings won't produce this anyway, but sometimes similar headings can be legitimate. I'm thinking of a tutorial where you might have:

:: software 1

::: get sources

::: build

:: software 2

::: get sources

::: build

Here it would be a problem. But I agree that a counter is not a good solution actually.

**Comment by Ypnose** [comment]
Indeed, that is a good example. I'm thinking about a "fix". As each heading creates a specific section, appending a counter could do the trick.
